processor/src/main/java/com/squareup/sqlbrite/SqlBriteProcessor.java

```diff
@@ -0,0 +1,152 @@
+import com.squareup.sqlbrite.annotation.Column;
+import com.squareup.sqlbrite.annotation.Table;
+
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.Filer;
+import javax.annotation.processing.Messager;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.annotation.processing.RoundEnvironment;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.PackageElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.util.ElementFilter;
+import javax.tools.Diagnostic;
+
+public class SqlBriteProcessor extends AbstractProcessor {
+  private static ClassName CONTENT_VALUES_CLASS = ClassName.get("android.content", "ContentValues");
+
+  private Filer filer;
+  private Messager messager;
+
+  @Override
+  public synchronized void init(ProcessingEnvironment processingEnv) {
+    super.init(processingEnv);
+    filer = processingEnv.getFiler();
+    messager = processingEnv.getMessager();
+  }
+
+  @Override
+  public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+    for (Element annotated : roundEnv.getElementsAnnotatedWith(Table.Builder.class)) {
+      processType(annotated);
+    }
+    return false;
+  }
+
+  private void processType(Element element) {
+    if (element.getKind() != ElementKind.INTERFACE) {
+      messager.printMessage(Diagnostic.Kind.ERROR, "Can only annotate " + Table.Builder.class +
+          " on interfaces.", element);
+      return;
+    }
+
+    TypeElement typeElement = (TypeElement) element;
+    if (typeElement.getModifiers().contains(Modifier.PRIVATE)) {
+      messager.printMessage(Diagnostic.Kind.ERROR, "Can't annotate private interfaces with " +
+          Table.Builder.class, element);
+      return;
+    }
+
+    String builderName = classNameOf(typeElement).replaceAll("\\.", "_") + "Impl";
+    TypeSpec.Builder builder = TypeSpec.classBuilder(builderName)
+        .addSuperinterface(TypeName.get(typeElement.asType()));
+    FieldSpec field = FieldSpec.builder(CONTENT_VALUES_CLASS, "values")
+        .initializer("new $T()", CONTENT_VALUES_CLASS).build();
+    builder.addField(field);
+
+    String packageName = packageNameOf(typeElement);
+    ClassName builderClassName = ClassName.get(packageName, builderName);
+    for (ExecutableElement method : ElementFilter.methodsIn(typeElement.getEnclosedElements())) {
+      final Column column = method.getAnnotation(Column.class);
+      if (column != null) {
+        String columnName = column.value();
+        List<? extends VariableElement> params = method.getParameters();
+        if (params.size() == 1) {
```

Comment:
We'll need to add a bit more validation here on the methods. I'd like to ensure that a `build` method is present and fail when methods take the wrong type as a parameter or have the wrong parameters. Doesn't need to happen in this first commit.